### DISCUSSION TITLE
Debug LDAP inside Nextcloud Container

### DISCUSSION THREAD

**Discussion by sim-san** [nextcloud/all-in-one#694] [Questions] [answered]
Is there a way to install ldap-utils inside the nextcloud Container ?

I want to test the connection from the nextcloud Container to an LDAP server running in another container. But I saw that the root user get`s an random password.

**Comment by szaimen** [comment, marked as answer]
simply run `sudo docker exec --user root -it nextcloud-aio-nextcloud bash` and then `apk add ...`?

**Reply by szaimen** [reply]
also, did you see https://github.com/nextcloud/all-in-one#ldap?

> The easiest way is by adding the LDAP docker container to the docker network nextcloud-aio. Then you can connect to the LDAP container by its name from the Nextcloud container.

**Reply by sim-san** [reply]
yes, I am working on this

**Reply by sim-san** [reply]
> also, did you see https://github.com/nextcloud/all-in-one#ldap?
> 
> > The easiest way is by adding the LDAP docker container to the docker network nextcloud-aio. Then you can connect to the LDAP container by its name from the Nextcloud container.

I working on this. The connection between the containers works fine. But I can't connect to internet anymore from the ldap container. This are my changes to the docker compose file of the ldap image:

```
xxx
    networks:
      - nextcloud-aio-network

xxx

networks:
  nextcloud-aio-network:
    external:
      name: nextcloud-aio
```
Is this right ?

**Reply by szaimen** [reply]
Looks fine to me. But I don't see how this should cut the connection of the ldap container to the internet...

**Reply by sim-san** [reply]
Should I add `host` to the list of networks ? Do you think that helps ?

**Comment by sim-san** [comment]
```
ncadmin@vm-nextcloud:~$ sudo docker exec --user root -it nextcloud-aio-nextcloud bash
bash-5.1# apk --update add openldap‑clients
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  openldap‑clients (no such package):
    required by: world[openldap‑clients]
```

Do you unterstand, why this package for ldapsearch [`openldap‑clients`](https://pkgs.alpinelinux.org/package/edge/main/x86_64/openldap-clients) is not found in the nextcloud container ?

**Reply by sim-san** [reply]
Also adding the repo does not help:
```
bash-5.1# cat /etc/apk/repositories
https://dl-cdn.alpinelinux.org/alpine/v3.15/main
https://dl-cdn.alpinelinux.org/alpine/v3.15/community
@edge http://dl-cdn.alpinelinux.org/alpine/edge/mai
```

```
bash-5.1# apk add --update openldap‑clients@edge
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  openldap‑clients (no such package):
    required by: world[openldap‑clients]
```

**Reply by szaimen** [reply]
wrong hyphen?

try this one: `openldap-clients`
